### PR TITLE
[ISSUE #421] feat: support ipv6

### LINKED ISSUE #421: [Native] support ipv6
The issue tracker is **ONLY** used for the go client (feature request of RocketMQ need to follow [RIP process](https://github.com/apache/rocketmq/wiki/RocketMQ-Improvement-Proposal)). Keep in mind, please check whether there is an existing same report before your raise a new one.

Alternately (especially if your communication is not a bug report), you can send mail to our [mailing lists](http://rocketmq.apache.org/about/contact/). We welcome any friendly suggestions, bug fixes, collaboration, and other improvements.

Please ensure that your bug report is clear and that it is complete. Otherwise, we may be unable to understand it or to reproduce it, either of which would prevent us from fixing the bug. We strongly recommend the report(bug report or feature request) could include some hints as to the following:

**BUG REPORT**  
**Please add the branch name [Native]/[Master] at the header of the Isssue title.**

1. Please describe the issue you observed:

    - What did you do (The steps to reproduce)?

    - What did you expect to see?

    - What did you see instead?

2. Please tell us about your environment:

     - What is your OS?

     - What is your client version?

     - What is your RocketMQ version?

3. Other information (e.g. detailed explanation, logs, related issues, suggestions on how to fix, etc):

**FEATURE REQUEST**

1. Please describe the feature you are requesting.

2. Provide any additional detail on your proposed use case for this feature.

2. Indicate the importance of this issue to you (blocker, must-have, should-have, nice-to-have). Are you currently using any workarounds to address this issue?

4. If there are some sub-tasks using -[] for each subtask and create a corresponding issue to map to the sub task:

    - [sub-task1-issue-number](example_sub_issue1_link_here): sub-task1 description here, 
    - [sub-task2-issue-number](example_sub_issue2_link_here): sub-task2 description here,
    - ...

### PR DESCRIPTION
**Pull request by xujianhai666**
- remove ipv4 check for nameserver
- add ipv6 check

CLoses #421

### CODE REVIEW

File: internal/remote/tcp_conn.go [hunk 1]
```diff
@@ -40,6 +48,19 @@ func initConn(ctx context.Context, addr string) (*tcpConnWrapper, error) {
 	}, nil
 }
 
+// reference by net.ParseIP
+func isIPV4(s string) bool {
+	for i := 0; i < len(s); i++ {
+		switch s[i] {
+		case '.':
+			return true
```
Comment (ShannonDing) [inline internal/remote/tcp_conn.go:56]: for a domain like www.nameserver.com:8080, shall we consider it as an IPv4 string?

Comment (xujianhai666) [reply internal/remote/tcp_conn.go:56]: we just justify ipv6 or  ipv6, domain name never runn into here

Comment (ShannonDing) [reply internal/remote/tcp_conn.go:56]: OK, Got it. 
The native branch has been moved to the master, you can submit this PR directly to the master。

**Comment by shenhui0509** [comment]
The `DecodeMessage` function should also be compatible with ipv6 host stored.
Referred to https://github.com/apache/rocketmq/blob/master/common/src/main/java/org/apache/rocketmq/common/message/MessageDecoder.java#L317

**Comment by wenfengwang** [comment]
close due to https://github.com/apache/rocketmq-client-go/pull/641 and long time no active. if need, pls reopen again

**Pull request closed without merging**